### DISCUSSION TITLE
How to handle TIFF Image with OJPEG Compression

### DISCUSSION THREAD

**Discussion by tonyfkchu** [dlemstra/Magick.NET#660] [Help] [answered]
Dear Expert,

I have running into a issue for readinging Tiff file by

IMImage = New MagickImage(image)

and the Error Message is

"ImageMagick Exception: Inconsistent number of MCU in codestream. `OJPEGDecodeRaw' @ error/tiff.c/TIFFErrors/661"

Any Suggestion to solve this issue?

Many Thanks

Tony

**Comment by dlemstra** [comment]
The most likely reason that this is happening because the file is incorrect/corrupt, Is it possible to share the file here?

**Comment by tonyfkchu** [comment]
Here is the image 1 sample and actually those set of image have the same problem 
https://app.box.com/s/7ws3zci62vl8ki9caeyp8eqg420lwtb1

The link have been removed

**Reply by dlemstra** [reply]
That file is private?

**Comment by tonyfkchu** [comment, marked as answer]
The New Link with less sensitive data https://app.box.com/s/l7h12wwjtp1rhrt3juej3zmnvbxep33n

The link have been removed

**Reply by dlemstra** [reply]
That image contains a finger print? Not sure if it is a real or fake file though.

It seems that due to updating the libtiff library a check was added that broke reading OJPEG files. This patch fixed it: https://gitlab.com/libtiff/libtiff/-/commit/0356ea76bac908c61160d735f078437ace953bd3. We keep a copy of the libtiff library and I updated that file to the latest version in the repository of libtiff and that resolves your issue. Will publish a new release later this week and then you will be able to read this file.

**Reply by tonyfkchu** [reply]
Many Thanks! and Please update here if the new release have been uploaded

**Reply by dlemstra** [reply]
The new release (7.19.0.1) has been published.
